File: leetcode/easy/35_搜索插入位置/35_搜索插入位置.py
#二分法
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        n = len(nums)
        left = 0
        right = n - 1
        while left <= right:
            middle = left + ((right - left) >> 1)
            if nums[middle] > target:
                right = middle - 1
            elif nums[middle] < target:
                left = middle + 1
            else:
                return middle
        return right + 1

File: leetcode/easy/35_搜索插入位置/35_搜索插入位置.go
//二分法
func searchInsert(nums []int, target int) int {
    n := len(nums)
    left := 0
    right := n - 1
    for left <= right {
        middle := left + ((right - left) >> 1)
        if nums[middle] > target {
            right = middle - 1
        } else if nums[middle] < target {
            left = middle + 1
        } else {
            return middle
        }
    }
    return right + 1
}